geetools/ee_list.py

```diff
@@ -287,3 +287,46 @@ def zip(self) -> ee.List:
         """
         indices = ee.List.sequence(0, ee.List(self._obj.get(0)).size().subtract(1))
         return indices.map(lambda i: self._obj.map(lambda j: ee.List(j).get(i)))
+
+    def chunked(self, parts: int | ee.Number) -> ee.List:
+        """Break a :py:class:`ee.List` into lists of length `parts`.
+
+        Args:
+            parts: the number of parts to get.
+
+        Returns:
+            a list of lists.
+
+        Examples:
+            .. jupyter-execute::
+
+                import ee, geetools
+                from geetools.utils import initialize_documentation
+
+                initialize_documentation()
+
+                l = ee.List([1,2,3,4,5,6,7,8,9])
+                l = l.geetools.chunked(3)
+                l.getInfo()
+        """
+        # parse the user parameters
+        parts = ee.Number(parts).toInt()
+        totalSize = self._obj.size()
+
+        # create a list of list of fixed sized elements corresponding of the
+        # euclidean division of the object length by the number of parts
+        size = totalSize.divide(parts).floor()
+        idxList = ee.List.sequence(0, parts.subtract(1))
+
+        def compute_parts(idx: ee.Number) -> ee.List:
+            start = ee.Number(idx).multiply(parts)
```

Comment:
small mistake with big consequences, here it should be multiplied by size and not part. It was invisible in your example because size and part are the same (3). split it inot 4 and and "3" and "4" will be dropped.

Reply:
https://github.com/gee-community/geetools/pull/439/commits/7d6041240cc665171ff08f35a71b65ee268ecdb9